### ISSUE TITLE
[BUG] dataset.to_ddf().compute() throws an error on parquet files from pyarrow/fastparquet

### ISSUE THREAD

**Issue by benfred** [NVIDIA/NVTabular#411]
**Describe the bug**
Calling ```dataset.to_ddf().compute()``` throws an exception when run on parquet datasets written out by pyarrow or fastparquet. The same dataset written out by cudf doesn't have an issue , nor do we have an issue when setting the columns to load (via the columns param in to_ddf).

<details><summary>Full error here:</summary>
<p>

```
Traceback (most recent call last):
  File "broken2.py", line 15, in <module>
    nvtabular.Dataset(pyarrow_path).to_ddf().compute()
  File "/home/ben/anaconda3/envs/cudf_dev/lib/python3.7/site-packages/dask/base.py", line 167, in compute
    (result,) = compute(self, traverse=False, **kwargs)
  File "/home/ben/anaconda3/envs/cudf_dev/lib/python3.7/site-packages/dask/base.py", line 452, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/home/ben/anaconda3/envs/cudf_dev/lib/python3.7/site-packages/dask/local.py", line 527, in get_sync
    return get_async(apply_sync, 1, dsk, keys, **kwargs)
  File "/home/ben/anaconda3/envs/cudf_dev/lib/python3.7/site-packages/dask/local.py", line 471, in get_async
    fire_task()
  File "/home/ben/anaconda3/envs/cudf_dev/lib/python3.7/site-packages/dask/local.py", line 466, in fire_task
    callback=queue.put,
  File "/home/ben/anaconda3/envs/cudf_dev/lib/python3.7/site-packages/dask/local.py", line 516, in apply_sync
    res = func(*args, **kwds)
  File "/home/ben/anaconda3/envs/cudf_dev/lib/python3.7/site-packages/dask/local.py", line 227, in execute_task
    result = pack_exception(e, dumps)
  File "/home/ben/anaconda3/envs/cudf_dev/lib/python3.7/site-packages/dask/local.py", line 222, in execute_task
    result = _execute_task(task, data)
  File "/home/ben/anaconda3/envs/cudf_dev/lib/python3.7/site-packages/dask/core.py", line 121, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "/home/ben/anaconda3/envs/cudf_dev/lib/python3.7/site-packages/dask/dataframe/io/parquet/core.py", line 287, in read_parquet_part
    df = df[[c for c in columns if c not in index]]
  File "/home/ben/anaconda3/envs/cudf_dev/lib/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "/home/ben/code/cudf/python/cudf/cudf/core/dataframe.py", line 677, in __getitem__
    return self._get_columns_by_label(mask)
  File "/home/ben/code/cudf/python/cudf/cudf/core/frame.py", line 443, in _get_columns_by_label
    new_data = self._data.select_by_label(labels)
  File "/home/ben/code/cudf/python/cudf/cudf/core/column_accessor.py", line 214, in select_by_label
    return self._select_by_label_list_like(key)
  File "/home/ben/code/cudf/python/cudf/cudf/core/column_accessor.py", line 261, in _select_by_label_list_like
    to_flat_dict({k: self._grouped_data[k] for k in key}),
  File "/home/ben/code/cudf/python/cudf/cudf/core/column_accessor.py", line 261, in <dictcomp>
    to_flat_dict({k: self._grouped_data[k] for k in key}),
KeyError: None
```
</p>
</details>

**Steps/Code to reproduce bug**
```python
import cudf
import pandas as pd
import nvtabular

# define a simple dataframe
df = pd.DataFrame({"col1": [0] * 25})

# if we write it out with cudf we can read it back in no problem with a NVT dataset
cudf_path = "cudf.parquet"
cudf.from_pandas(df).to_parquet(cudf_path)
nvtabular.Dataset(cudf_path).to_ddf().compute()

# if we write this dataframe as parquet with pyarrow or fastparquet, we can't read it back in unless we specify the columns
pyarrow_path = "pyarrow.parquet"
df.to_parquet(pyarrow_path, engine="pyarrow")
nvtabular.Dataset(pyarrow_path).to_ddf().compute().  # <--- throws an exception
```

**Comment by benfred** [comment]
For what it's worth, we can get the same error in dask_cudf when reading in these files - so this might be an issue in dask_cudf.

For instance this calll fails on the parquet file generated from pyarrow:
```python
dask_cudf.read_parquet(pyarrow_path, index=False).compute()
```

While the same call works on the parquet file generated from cudf:
```python
dask_cudf.read_parquet(cudf_path, index=False).compute()
```

(cudf.read_parquet can read in both versions)

@rjzamora A simple workaround for us is to not set the ```index=False``` param when we don't have a list of columns supplied, but this might be something that needs fixed upstream?

**Comment by rjzamora** [comment]
Yup - This seems like a `dask_cudf` bug for sure.   Round-tripping a null-named index was a pain to handle in `dask.dataframe`, and it seems like something is getting screwed up for `dask_cudf`.   The good news for NVTabular is that the `index=False` is *probably* not necessary.  For now, we can do something like this to handle the special case that the pandas metadata lists a non-null index name:

```python
    def to_ddf(self, columns=None):
        ddf = dask_cudf.read_parquet(
            self.paths,
            columns=columns,
            gather_statistics=False,
            split_row_groups=self.row_groups_per_part,
            storage_options=self.storage_options,
        )
        if ddf.index.name is not None:
            return ddf.reset_index(drop=False)
        return ddf
```

**Comment by rjzamora** [comment]
Ah - Just saw that you already included a fix in #410 

I'll plan to raise a separate dask_cudf issue and get a fix in upstream too.

**Comment by benfred** [comment]
We have a workaround in #410
